packages/build/src/plugins/child/mutations.js

```diff
@@ -202,6 +202,18 @@ const setBuildCommandOrigin = function (topProxy) {
   topProxy.build.commandOrigin = 'plugin'
 }
 
+const setTopFunctionsDirectory = function (topProxy, value, key) {
+  setFunctionsDirectory(topProxy, value)
+  setFunctionsCatchAll(topProxy, value, key)
+}
+
+// When settings `functions.{propName}`, we also set `functions.*.{propName}`,
```

Comment:
Sorry for the late review, but just wondering if this accounts for an edge case in which a function has the same name as one of the config properties (e.g. `included_files`). If someone does `functions.included_files.node_bundler`, will this set an `included_files` property on `functions['*']`, effectively setting the `included_files` for all functions with an invalid value?

Reply:
Oh, I forgot about this, good point.
I am planning on trying to refactor the whole file so we call `@netlify/config` instead for parts of the logic, which should solve this issue.

Reply:
Sounds good!